reductus/sansred/steps.py

```diff
@@ -2046,3 +2091,212 @@ def getPoissonUncertainty(y):
+
+    **Inputs**
+
+    data (sans1d[]): SANS 1D data reduced data
+
+    **Returns**
+
+    output (sans1d): A combined reduced data set with all points stitched together into a single data object
+
+    | 2025-07-29 Jeff Krzywon initial implementation
+    """
+    # Create a data object, so we aren't modifying the underlying data that will be displayed
+    scaled_data = SansIQData(np.zeros(0), np.zeros(0), np.zeros(0), np.zeros(0), np.zeros(0), np.zeros(0))
+    for datum in data:
+        # Get data with the q points cutoff
+        cutoff = datum.q_cutoff()
+        # Scale the intensity by the scaling factor
+        new_data = rescale_1d(cutoff, datum.scaling_factor)
+        scaled_data.append_1d_data_set(new_data)
+    return scaled_data
+
+
+def _find_nearest(array: np.ndarray, value: float) -> float:
+    """Find the value in an array closest to the seek value.
+
+    :param array: A numpy array of numerical values any dimension.
+    :param value: The value to compare to the array used to find the closest array value.
+    :return: The array value closest to the seek value.
+    """
+    array = np.asarray(array)
+    idx = (np.abs(array - value)).argmin()
+    return array[idx]
+
+
+def _get_scaling_factor_between_data(scale_to_data: np.ndarray, data_to_be_scaled: np.ndarray) -> float:
+    """Find the overlap region between two numerical arrays, and find the ratio of the sum of values in the overlap
+    region.
+
+    :param scale_to_data: A numerical array
+    """
+
+    if min(data_to_be_scaled) > min(scale_to_data):
+        # Overlap is on the high-Q end of the data to be scaled
+        scale_to_upper_index = np.where(scale_to_data == _find_nearest(scale_to_data, np.max(data_to_be_scaled)))[0]
+        scaled_lower_index = np.where(data_to_be_scaled == _find_nearest(data_to_be_scaled, np.min(scale_to_data)))[0]
+        numerator = np.sum(data_to_be_scaled[scaled_lower_index:])
+        denominator = np.sum(scale_to_data[:scale_to_upper_index])
+    else:
+        # Overlap is on the low-Q end of the data to be scaled
+        scaled_upper_index = np.where(scale_to_data == _find_nearest(scale_to_data, np.min(data_to_be_scaled)))[0]
+        scale_to_lower_index = np.where(data_to_be_scaled == _find_nearest(data_to_be_scaled, np.max(scale_to_data)))[0]
+        numerator = np.sum(data_to_be_scaled[:scaled_upper_index])
+        denominator = np.sum(scale_to_data[scale_to_lower_index:])
+    # Slice data at limits after determining if you slice the upper or lower section
+    # Sum all values in the sliced section
+    # Divide scale_to_data sum by data_to_be_scaled sum
+    return numerator / denominator
+
+
+def _get_shifting_factor_between_data(scale_to_data: np.ndarray, data_to_be_scaled: np.ndarray) -> float:
+    """Find the overlap region between two numerical arrays, and average difference between the regions
+
+    :param scale_to_data: A numerical array
+    """
+
```

Comment:
This code is making the assumption that the q values are sorted. Using numpy.searchsorted is quicker and clearer.